lua/nvim-treesitter/parsers.lua

```diff
@@ -1,2618 +1,2565 @@
     tier = 2,
   },
-
-  wing = {
-    install_info = {
-      url = 'https://github.com/winglang/tree-sitter-wing',
-      files = { 'src/parser.c', 'src/scanner.c' },
-    },
-    maintainers = { '@gshpychka', '@MarkMcCulloh' },
-    tier = 3,
-  },
-
   wgsl = {
     install_info = {
-      url = 'https://github.com/szebniok/tree-sitter-wgsl',
       files = { 'src/parser.c', 'src/scanner.c' },
+      revision = '40259f3c77ea856841a4e0c4c807705f3e4a2b65',
+      url = 'https://github.com/szebniok/tree-sitter-wgsl',
     },
     maintainers = { '@szebniok' },
     tier = 3,
   },
-
   wgsl_bevy = {
     install_info = {
-      url = 'https://github.com/tree-sitter-grammars/tree-sitter-wgsl-bevy',
       files = { 'src/parser.c', 'src/scanner.c' },
       generate_from_json = true,
+      revision = 'f5980f534ee64256b1e64b0a42e2864d91154a5d',
+      url = 'https://github.com/tree-sitter-grammars/tree-sitter-wgsl-bevy',
     },
     maintainers = { '@theHamsta' },
     tier = 2,
   },
-
+  wing = {
+    install_info = {
+      files = { 'src/parser.c', 'src/scanner.c' },
+      revision = 'bd1d35cf3e013dc7e189b46a593bdc2b281b0dd7',
+      url = 'https://github.com/winglang/tree-sitter-wing',
+    },
+    maintainers = { '@gshpychka', '@MarkMcCulloh' },
+    tier = 3,
+  },
   wit = {
     install_info = {
-      url = 'https://github.com/liamwh/tree-sitter-wit',
       files = { 'src/parser.c' },
+      revision = 'ed13e0a3ee77833900a095d57d2685a1f7f91237',
+      url = 'https://github.com/liamwh/tree-sitter-wit',
     },
     maintainers = { '@liamwh' },
     tier = 3,
   },
-
   xcompose = {
     install_info = {
-      url = 'https://github.com/tree-sitter-grammars/tree-sitter-xcompose',
       files = { 'src/parser.c' },
+      revision = '2383cc69a2c42cfade41c7cb971fb3862bec6df1',
+      url = 'https://github.com/tree-sitter-grammars/tree-sitter-xcompose',
     },
     maintainers = { '@ObserverOfTime' },
     tier = 2,
   },
-
   xml = {
     install_info = {
-      url = 'https://github.com/tree-sitter-grammars/tree-sitter-xml',
       files = { 'src/parser.c', 'src/scanner.c' },
       location = 'xml',
+      revision = '96beba9f3e4156aaae64bb86997503b1ceb1283d',
+      url = 'https://github.com/tree-sitter-grammars/tree-sitter-xml',
     },
     maintainers = { '@ObserverOfTime' },
     requires = { 'dtd' },
     tier = 2,
   },
-
   yaml = {
     install_info = {
-      url = 'https://github.com/tree-sitter-grammars/tree-sitter-yaml',
       files = { 'src/parser.c', 'src/scanner.c' },
+      revision = 'ee093118211be521742b9866a8ed8ce6d87c7a94',
+      url = 'https://github.com/tree-sitter-grammars/tree-sitter-yaml',
     },
     maintainers = { '@amaanq' },
     tier = 2,
   },
-
   yang = {
     install_info = {
-      url = 'https://github.com/Hubro/tree-sitter-yang',
       files = { 'src/parser.c' },
+      revision = '2c0e6be8dd4dcb961c345fa35c309ad4f5bd3502',
+      url = 'https://github.com/Hubro/tree-sitter-yang',
```

Comment:
we had a similar PR before. Wouldn't it be simpler to separate human edited data from machine generated files? The parser file needs to be reloaded in order to make TSUpdate on plugin update to work.

Reply:
The problem is that the current setup mixes that as well, with the revision field. And we don't support hot-reloading; but if this breaks TSUpdate completely, I will have to work around that, of course.

Reply:
And we are actively moving away from tracking head; this is in preparation for a version field.

Reply:
We could (and arguably should) make :TSUpdate invalidate the parsers module. Since it doesn't contain anything except the data anymore, that should be fine.

Reply:
Ah, but that would break the `require('nvim-treesitter.parsers').zimbu.install_info.revision = <sha>` use case...

To be clear, I'm not 100% convinced of this either, but it does simplify the code, consolidate sources of truth, and aligns with the general "if you update plugins, you need to reload" story.